tools/dev.py

```diff
@@ -21,205 +24,256 @@ class Colour:
-migr_apply_parser.add_argument("--create-data", action="store_true",
-                               help="create some random data to work on like conference, buses, "
-                                    "rooms, etc.")
-migrate_subparsers.add_parser("make", aliases=["m"], add_help=False,
-                              help=f"generate Django migrations from models {FILE_SYSTEM_NOTE}")
-
-run_server_parser = subparsers.add_parser(
-    "server", aliases=["sv"],
-    help="run Django development server inside the container (localhost, port 8000)")
-
-web_parser = subparsers.add_parser(
-    "web", aliases=["javascript", "js"],
-    help=f"perform action related to web application {SUBCOMMANDS_NOTE}")
-
-web_subparsers = web_parser.add_subparsers(dest="action", metavar="ACTION", required=True)
-
-web_subparsers.add_parser("install", aliases=["i"], add_help=False,
-                          help="install web app's dependencies from file `package.json`")
-web_subparsers.add_parser("build", aliases=["b"], add_help=False,
-                          help=f"build web app {FILE_SYSTEM_NOTE}")
-web_subparsers.add_parser("watch", aliases=["w"], add_help=False,
-                          help=f"rebuild web app on file change {FILE_SYSTEM_NOTE}")
-
-py_parser = subparsers.add_parser("python", aliases=["py"],
-                                  help=f"perform action related to Python language "
-                                       f"{SUBCOMMANDS_NOTE}")
-
-py_subparsers = py_parser.add_subparsers(dest="action", metavar="ACTION", required=True)
-
-py_subparsers.add_parser("install", aliases=["i"], add_help=False,
-                         help="install Python dependencies from file `requirements.txt`")
-py_subparsers.add_parser("upgrade", aliases=["u"], add_help=False,
-                         help="upgrade Python dependencies from file `requirements.txt`")
-py_subparsers.add_parser("pip", add_help=False, help="upgrade pip")
-
-args = parser.parse_args()
-
-if args.command in ["run", "r"]:
-    print(f"{Colour.BLUE}-- Setup container --{Colour.NORMAL}")
-    setup(args.no_cache)
-    print(f"{Colour.BLUE}-- Run migrations --{Colour.NORMAL}")
-    migrate(args.create_admin, args.create_data)
-    print(f"{Colour.BLUE}-- Run webserver --{Colour.NORMAL}")
-    run_server()
-elif args.command in ["start", "setup", "s"]:
-    setup(args.no_cache)
-elif args.command in ["quit", "shutdown", "q"]:
-    docker_compose_run(["down"])
-elif args.command in ["test", "t"]:
-    docker_python(["test"] + (["-v", "2"] if args.verbose else []))
-elif args.command in ["shell", "sh"]:
-    if args.shell in ["bash"]:
-        docker_exec(["/bin/bash"], WEB_CONTAINER_NAME)
-    elif args.shell in ["postgres", "psql"]:
-        docker_exec(["psql", "-U", "zosia"], DB_CONTAINER_NAME)
-    else:
-        shell_parser.print_help()
-elif args.command in ["migrations", "m"]:
-    if args.action in ["apply", "a"]:
+def cli():
```

Comment:
If you'd like to enclose it in a function, you should split the new function into parser generation and argument deduction.
```
def cli():
    parser = create_parser()
    args = parser.parse_args()
    deduce_args(args)
```

Reply:
Splitting it as proposed has one major issue - we would have to return multiple parsers in order to handle help printing for subparsers :confused: 

```
    else:
          migrate_parser.print_help()
```